homework/shared_ptr/weak_ptr.hpp

```diff
@@ -89,4 +84,23 @@ weak_ptr<T>& weak_ptr<T>::operator=(weak_ptr<T>&& previousOwner) noexcept {
     return *this;
 }
 
+template <typename T>
+void weak_ptr<T>::reset() noexcept {
+    if (!counter_->getWeakRefs()) {
+        delete ptr_;
+    } else {
+        ptr_ = nullptr;
+    }
+}
+
+template <typename T>
+bool weak_ptr<T>::expired() const noexcept {
+    return counter_ ? counter_->expired() : false;
```

Comment:
Flawed. If this weak_ptr is not associated with any ptr(and therefore counter) it should also return that it is expired. counter_ itself doesn't have to be expired for this weak_ptr to be

Reply:
expired should return true i_f the managed object has already been deleted and is Equivalent to use_count() == 0 as in cpp.reference.
return should be like ```return counter_->expired()``` and this method check shared_refs count

Reply:
https://onlinegdb.com/BJJoXw4Uv

Reply:
same test add, fixes all test pass.